### ISSUE TITLE
[feature] Pragma/feature to disable specialness of $a, $b

### ISSUE THREAD

**Issue by t-a-k** [Perl/perl5#18251]
As of now, package variables `$a` and `$b` are exempt from "used only once" warnings and "requires explicit package name" errors.
It would be nice to have a pragma or feature to turn off this special treatment so that we can find (accidental) misuse of these variables.

Even for programs using `sort`, we can use `our` to suppress these diagnostics (or can use prototyped subroutines), or, I think it would be more convenient if these diagnostics are automatically disabled in the block just after `sort`.

**Comment by Grinnz** [comment]
Unfortunately the exemption can't be limited to sort because `$a` and `$b` can also be special in this way in subroutines passed to sort which have their own scope.

I think it would be a nice feature but in the meantime https://metacpan.org/pod/Perl::Critic::Policy::Freenode::DollarAB may help you find misuse of them.

**Comment by t-a-k** [comment]
> I think it would be a nice feature but in the meantime https://metacpan.org/pod/Perl::Critic::Policy::Freenode::DollarAB may help you find misuse of them.

DollarAB seems to be useful, but for programs that does not use `sort`, I think this is a bit overkill because `$a` and `$b` are popular names for temporary variables.

For example, in blead's `t/op/inc.t`, `$a` and `$c` are declared with `my`, but `$b` is used without any declaration while `use strict` is in effect.  I don't know whether this is intentional or not, but it would be nice if this could be detected.
Thank you anyway.
